quickfixj-core/src/test/java/quickfix/SocketInitiatorTest.java

```diff
@@ -145,7 +145,7 @@ public void testBlockLogoffAfterLogon() throws Exception {
                 clientApplication.stopAfterLogon(initiator);
                 clientApplication.setUpLogonExpectation();
 
-                initiator.block();
+                initiator.start();
```

Comment:
Doesn't this make for a potiential race condition between the initiator and the line below?

Reply:
Not really, it will simply block until it has actually logged in, tests passed twice after this change, on my PC and Jenkins.

Reply:
Also, the reason for the change was the deprecation of the `block()` method:
```
    @Override
    @Deprecated
    /**
     * Use SocketInitiator.start().
     */
    public void block() throws ConfigError, RuntimeError {
        initialize(false);
    }
```

Reply:
Isn't the point of this code to test the deprecated functionality. This test no longer tests what it says it tests.  It says it's testing `block()`.  You're removing test coverage for deprecated, yet still functional code. We should still test APIs we deprecate yet don't remove.

I would assume `start()` doesn't block and `block()` does. That would be the whole point, no? So you then have a race between start calling logoff and initiator.isLoggedOn() being true.

Reply:
You may have a point, let me undo this part.